From cd27c8faca0eeb0d4309cc5837dfdf3c42eba4e7 Mon Sep 17 00:00:00 2001
From: Armin Novak <armin.novak@thincast.com>
Date: Tue, 10 Mar 2026 09:17:23 +0100
Subject: [PATCH] [codec,planar] add early length check to avoid oob read

---
 libfreerdp/codec/planar.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libfreerdp/codec/planar.c b/libfreerdp/codec/planar.c
index ff161bfb75b2..fd2db2f44e28 100644
--- a/libfreerdp/codec/planar.c
+++ b/libfreerdp/codec/planar.c
@@ -738,9 +738,9 @@ BOOL freerdp_bitmap_decompress_planar(BITMAP_PLANAR_CONTEXT* WINPR_RESTRICT plan
 
 	const BYTE* srcp = pSrcData;
 
-	if (!pSrcData)
+	if (!pSrcData || (SrcSize < 1))
 	{
-		WLog_ERR(TAG, "Invalid argument pSrcData=nullptr");
+		WLog_ERR(TAG, "Invalid argument pSrcData=%p [size=%" PRIu32 "]", pSrcData, SrcSize);
 		return FALSE;
 	}
 
